app/lib/grades/grades_service/grades_service.dart

```diff
@@ -640,6 +647,7 @@ class SubjectResult extends Equatable {
   final Design design;
   final IList<ConnectedCourse> connectedCourses;
   final GradeTypeId finalGradeTypeId;
+  final Weight weightingForTermGrade;
```

Comment:
Is this better for the UI or would you rather have `IMap<SubjectId, Weight> TermResult.subjectWeights` or sth similar?